File: server/internal/model/meeting.go
package model

import "time"

type Meeting struct {
	ID        string    `json:"id"`
	Title     string    `json:"title"`
	Password  string    `json:"-"`
	HostID    string    `json:"host_id"`
	Status    string    `json:"status"` // pending, active, ended
	CreatedAt time.Time `json:"created_at"`
}

type Participant struct {
	ID        string `json:"id"`
	MeetingID string `json:"meeting_id"`
	UserID    string `json:"user_id"`
	Role      string `json:"role"` // host, participant
}

File: client/lib/models/meeting.dart
class Meeting {
  final String id;
  final String title;
  final String status;

  Meeting({
    required this.id,
    required this.title,
    required this.status,
  });

  factory Meeting.fromJson(Map<String, dynamic> json) {
    return Meeting(
      id: json['id'] ?? '',
      title: json['title'] ?? '',
      status: json['status'] ?? 'pending',
    );
  }
}

class Participant {
  final String id;
  final String meetingId;
  final String userId;
  final String role;

  Participant({
    required this.id,
    required this.meetingId,
    required this.userId,
    required this.role,
  });

  factory Participant.fromJson(Map<String, dynamic> json) {
    return Participant(
      id: json['id'] ?? '',
      meetingId: json['meeting_id'] ?? '',
      userId: json['user_id'] ?? '',
      role: json['role'] ?? 'participant',
    );
  }
}